packages/react-instantsearch-core/src/types.ts

```diff
@@ -0,0 +1,12 @@
+// from AlgoliaSearchHelper:
+export type SearchParameters = any; // AlgoliaSearchHelper.SearchParameters
+export type SearchResults = any; // AlgoliaSearchHelper.SearchResults
+export type SearchForFacetValuesResults = any;
+
+// internal
+type UIState = {
+  [index: string]: any;
+};
+export type SearchState = { indices?: UIState } | UIState;
+
+export type MetaData = any[];
```

Comment:
```suggestion
export type SearchMetadata = any;
```

We can then compose the type to create multiple metadata `type SearchMetadatas = SearchMetadata[];`.